types/object-hash/index.d.ts

```diff
@@ -1,42 +1,104 @@
+//                 Artur Diniz <https://github.com/artdiniz>
+//                 Martin Badin <https://github.com/martin-badin>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
-interface IStream {
-    update?(chunk: any, encoding: BufferEncoding, callback: (error?: Error | null) => void): void;
-    write?(chunk: any, encoding: BufferEncoding, callback: (error?: Error | null) => void): void;
-}
-type BufferEncoding = "ascii" | "utf8" | "utf-8" | "utf16le" | "ucs2" | "ucs-2" | "base64" | "latin1" | "binary" | "hex";
-import HashStatic = ObjectHash.HashStatic;
-export = HashStatic;
-export as namespace objectHash;
+type BufferEncoding =
+    | 'ascii'
+    | 'base64'
+    | 'binary'
+    | 'hex'
+    | 'latin1'
+    | 'ucs-2'
+    | 'ucs2'
+    | 'utf-8'
+    | 'utf16le'
+    | 'utf8';
 
 declare namespace ObjectHash {
-    export interface IOptions {
-        algorithm?: string;
-        encoding?: string;
+    interface Stream {
+        update?(chunk: any, encoding: BufferEncoding, callback: (error?: Error | null) => void): void;
+        write?(chunk: any, encoding: BufferEncoding, callback: (error?: Error | null) => void): void;
+    }
+
+    interface Options {
+        /**
+         * @default 'sha1'
+         */
+        algorithm?: 'sha1' | 'md5' | 'passthrough';
+        /**
+         * @default 'hex'
+         */
+        encoding?: 'buffer' | 'hex' | 'binary' | 'base64';
+
+        excludeKeys?: (key: string) => boolean;
+        /**
+         * @default false
+         */
         excludeValues?: boolean;
+        /**
+         * @default false
+         */
         ignoreUnknown?: boolean;
+
         replacer?: (value: any) => any;
-        respectFunctionProperties?: boolean;
+        /**
+         * @default true
+         */
         respectFunctionNames?: boolean;
+        /**
+         * @default true
+         */
+        respectFunctionProperties?: boolean;
+        /**
+         * @default true
+         */
         respectType?: boolean;
+        /**
+         * @default false
+         */
         unorderedArrays?: boolean;
-        unorderedSets?: boolean;
+        /**
+         * @default true
+         */
         unorderedObjects?: boolean;
-        excludeKeys?: (key: string) => boolean;
-    }
-
-    export interface Hash {
-        (object: any, options?: IOptions): string;
-        sha1(object: any): string;
-        keys(object: any): string;
-        MD5(object: any): string;
-        keysMD5(object: any): string;
-        writeToStream(value: any, stream: IStream): void;
-        writeToStream(value: any, options: IOptions, stream: IStream): void;
+        /**
+         * @default true
+         */
+        unorderedSets?: boolean;
     }
+}
 
-    export var HashStatic: Hash;
+interface ObjectHash {
+    /**
+     * @see https://github.com/puleos/object-hash#hashvalue-options
+     */
+    (object: any, options?: ObjectHash.Options): string;
```

Comment:
it should be anything, but not everything:
https://github.com/puleos/object-hash/blob/master/index.js#L23
this fails on unefined:
```js
Uncaught Error: Object argument required.
    at applyDefaults (C:\development\tests\tests-object-hash\node                                                              e_modules\object-hash\index.js:82:11)
```
works with all known types (JS that is), except of undefined, so `{} | null` (or something matching everything except of undefined)

Reply:
same for everything else in input IMO

Reply:
Thanks for your review, updated